### PR TITLE
Update Azure Communication Phone Numbers to 1.1.0-beta.3

### PR DESCRIPTION
**Pull request by whisper6284**
# Description

Update Azure Communication Phone Numbers to 1.1.0-beta.3.

This includes update to retry options in the client builder, and updates to the common and identity dependency packages.

### CODE REVIEW

**Comment by vcolin7** [comment]
We need to update the README as well to mention `1.1.0-beta.3`.

**Comment by whisper6284** [comment]
> 

I don't follow. I see the version on line 58 of the README already.

File: sdk/communication/azure-communication-phonenumbers/CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
 # Release History
 
-## 1.1.0-beta.3 (Unreleased)
+## 1.1.0-beta.3 (2022-03-16)
```
Comment (vcolin7) [inline sdk/communication/azure-communication-phonenumbers/CHANGELOG.md:3]: nit;
```suggestion
## 1.1.0-beta.3 (2022-03-11)
```

**Review by vcolin7 — APPROVED** [review]
LGTM

**Comment by vcolin7** [comment]
> > 
> 
> I don't follow. I see the version on line 58 of the README already.

Oh my bad, I must have been looking at a different PR at the time.

**Review by lucasrsant — APPROVED** [review]

**Pull request merged**